streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -43,10 +43,15 @@ class Checkpoint(@transient ssc: StreamingContext, val checkpointTime: Time)
   val delaySeconds = MetadataCleaner.getDelaySeconds(ssc.conf)
   val sparkConfPairs = ssc.conf.getAll
 
-  def sparkConf = {
-    new SparkConf(false).setAll(sparkConfPairs)
+  def createSparkConf(): SparkConf = {
+    val newSparkConf = new SparkConf(loadDefaults = false).setAll(sparkConfPairs)
       .remove("spark.driver.host")
       .remove("spark.driver.port")
+    val oldMaster = new SparkConf(loadDefaults = true).getOption("spark.master")
```

Comment:
Why is this being called `oldMaster`? Isn't this an option wrapping the new master?